src/main/java/pkmnman/ApplicationRunner.java

```diff
@@ -1,27 +1,161 @@
 package pkmnman;
 
 import pkmncore.Pokemon;
-import pkmncore.PokemonFinder;
+import pkmncore.PokemonError;
+import pkmncore.search.PokemonFinder;
+import pkmncore.storage.PokemonManager;
+
+import java.util.List;
 
 public class ApplicationRunner {
 
     private final PokemonFinder finder;
     private final Input input;
     private final Display display;
+    private final PokemonManager manager;
 
-    public ApplicationRunner(PokemonFinder finder, Input input, Display display) {
+    public ApplicationRunner(PokemonFinder finder, PokemonManager manager, Input input, Display display) {
         this.finder = finder;
+        this.manager = manager;
         this.input = input;
         this.display = display;
     }
 
     public void start() {
         display.clearScreen();
         display.greet();
+        getUserAction();
+    }
+
+    public void searchPage() {
+        display.clearScreen();
+        display.showSearchHeader();
         display.promptUser();
+        Pokemon pokemon = findPokemon();
+        searchedPokemonPage(pokemon);
+    }
+
+    private Pokemon findPokemon() {
+        String name = input.get();
+        Pokemon pokemon = finder.find(name);
+        display.clearScreen();
+        return pokemon;
+    }
+
+    public void addPage() {
+        display.clearScreen();
+        display.showAddHeader();
+        display.promptForPokemon();
         String name = input.get();
         Pokemon pokemon = finder.find(name);
+        pokemonPage(pokemon);
+    }
+
+    private void viewPage() {
+        display.clearScreen();
+        display.showViewHeader();
+        List<Pokemon> caughtPokemon = manager.viewCaughtPokemon();
+        displayCaughtPokemon(caughtPokemon);
+    }
+
+    private void getUserAction() {
+        display.showMenu();
+        String action = input.get().toUpperCase().trim();
+        switch (action) {
```

Comment:
Adding new actions will grow this switch-case statement. This means this function is not quite OCP compliant. Have you considered something like an "Action" class that would allow you to add new behaviour without special-casing it?